### ISSUE TITLE
[IDEA] Vanity Address Generator

### ISSUE THREAD

**Issue by Syth** [turtlecoin/meta#37]
I think it'd be pretty cool to have a vanity address generator for those that would want it. I wouldn't mind tackling the project if someone could point me in the right direction.

**Comment by nnamon** [comment]
Check out https://github.com/ehmry/nim-monero

**Comment by editfmah** [comment]
We have been discussing this, we will have to discuss with @RocksteadyTC as we might all have different preferences, but I see the value in distributing the registration of vanity addresses in the ledger and therefore making it entirely redundant.  We will then need to add RPC calls into the daemon to answer lookup requests.  Lots of the initial design work was already done for another project, although that does not mean any of it needs to be used.

My preference would be to allow registration from the wallet:

register `@napoleon`

That would deduct a charge (to be used for other things, maybe even dev fund or something), and add a sha512 hash in the tx extra.  Daemons would then have a global registry of all these registrations and could make a very quick lookup against it.

It shouldn't be that hard to prototype.

**Comment by RocksteadyTC** [comment]
@editfmah I'd be okay with this

**Comment by MoonMoonDogo** [comment]
I've forked the monero vanitygen and adapted it to TurtleCoin: https://github.com/MoonMoonDogo/turtle-vanitygen

There are a few limitations described in the README. I've tested the generated mnemonics with my decoder (https://github.com/MoonMoonDogo/paper-turtle) and got the same spend and view keys.
@ZedPea tested the mnemonics with his adapted simplewallet and also got the same key so I think we are fine.
